### PR TITLE
Allow users to specify different report templates

### PR DESCRIPTION
**Pull request by major**
Before this PR, `skt` only has the capability to send reports rendered from a single template. Users need the ability to specify different templates for different situations. For example:

* Some users may want to receive detailed reports, others may not
* Some reports might contain sensitive data (log files, URLs) that need to be restricted

This PR moves the old template to `base.j2`, where it can be included by various templates named `report_STRING.j2`. Those report files include `base.j2` and set variables that control the report output.

Two initial reports, `default` and `limited` are provided. The `limited` report is identical to `default`, but it excludes logs from hardware tests.

### CODE REVIEW

File: skt/templates/partials/test_failure.j2 [hunk 1]
```diff
@@ -11,10 +11,12 @@ One or more hardware tests failed:
   {% for test_result in job.test_results %}
     {% for failed_task in test_result.failed_tasks %}
     Test name: {{ failed_task.name }}
+      {% if not report_omit_hardware_test_logs %}
```
Comment (spbnick) [inline skt/templates/partials/test_failure.j2]: I don't like different parts of the report code communicating via global variables, but if there's no other option, so be it.

File: skt/executable.py [hunk 2]
```diff
@@ -814,6 +814,14 @@ def setup_parser():
         type=str,
         help='Use smtp url instead of localhost to send mail',
     )
+    parser_report.add_argument(
+        "--template",
+        dest="template",
+        type=str,
+        default='default',
+        choices=('default'),
```
[the remark this answers is not in the document]
Comment (major) [reply skt/executable.py]: Can do.

File: skt/reporter.py [hunk 3]
```diff
@@ -337,7 +337,8 @@ def _get_multireport(self):
         Returns: A long string of test results suitable for sending via email
                  or displaying directly in a terminal.
         """
-        template = JINJA_ENV.get_template('report.j2')
+        template_file = "report_{}.j2".format(self.cfg['template'])
```
[the remark this answers is not in the document]
Comment (major) [reply skt/reporter.py]: Makes sense.

File: skt/templates/partials/test_failure.j2 (hunk 1 above)
Comment (major) [reply skt/templates/partials/test_failure.j2]: That's the only option I've found so far. Let me know if you think of something else. ;)

**Pull request merged**